psiminer-core/src/main/kotlin/Pipeline.kt

```diff
@@ -13,15 +14,18 @@ import psi.transformations.PsiTreeTransformation
 import storage.Storage
 import java.io.File
 import java.util.concurrent.Executors
+import kotlin.io.path.Path
 
 class Pipeline(
     val language: Language,
     private val repositoryOpener: RepositoryOpener,
     psiTreeTransformations: List<PsiTreeTransformation>,
     private val filters: List<Filter>,
     val labelExtractor: LabelExtractor,
-    val storage: Storage
+    val storage: Storage,
+    val collectMetadata: Boolean = false
 ) {
+    private var metaDataStorage: MetaDataStorage? = null
```

Comment:
Maybe it's better to pass this storage into processPsiTree as an argument? Or it's too spaghetti?

Reply:
Methods in this class already have 2 pass-through arguments, so i don't think it's a good idea. 
(Actually i don't even know why existing arguments have to be pass-through and not pipeline arguments.)